tests/andromede/test_quota_co2.py

```diff
@@ -0,0 +1,194 @@
+    binding_constraints=[
+        Constraint(
+            name="Balance",
+            expression=port_field("FlowN", "F").sum_connections() == literal(0),
+        )
+    ],
+)
+
+"""
+Model of a simple power generator that takes account of CO² emissions related to the production.
+The power production p is bounded between p_min and p_max.
+An emission factor is used to determine the CO² emission according to the production.
+"""
+C02_POWER_MODEL = model(
+    id='CO2 power',
+    parameters=[float_parameter("p_min", CONSTANT),
+                float_parameter("p_max", CONSTANT),
+                float_parameter("cost", CONSTANT),
+                float_parameter("taux_emission", CONSTANT)],
+    variables=[float_variable("p", lower_bound=param("p_min"), upper_bound=param("p_max"))],
+    ports=[ModelPort(port_type=FLOW_PORT, port_name="FlowP"),
+           ModelPort(port_type=EMISSION_PORT, port_name="OutCO2")],
+    port_fields_definitions=[
+        PortFieldDefinition(
+            port_field=PortFieldId("FlowP", "F"),
+            definition=var("p"),
+        ),
+        PortFieldDefinition(
+            port_field=PortFieldId("OutCO2", "Q"),
+            definition=var("p") * param("taux_emission"),
+        )
+    ],
+    objective_contribution=(param("cost") * var("p")).sum().expec(),
+)
+
+
+"""
+Basic energy consumption model.
+It consume a fixed amount of energy "d" each hour.
+"""
+DEMAND_MODEL = model(
+    id='Demand model',
+    parameters=[float_parameter("d", CONSTANT)],
+    ports=[ModelPort(port_type=FLOW_PORT, port_name="FlowD")],
+    port_fields_definitions=[
+        PortFieldDefinition(
+            port_field=PortFieldId("FlowD", "F"),
+            definition=-param("d"),
+        )
+    ]
+)
+
+"""
+Model of the CO² quota. 
+It takes a set a CO² emissions as input. It forces the sum of those emissions to be smaller than a predefined quota. 
+"""
+QUOTA_CO2_MODEL = model(
+    id='QuotaCO2',
+    parameters=[float_parameter("quota", CONSTANT)],
+    ports=[ModelPort(port_type=EMISSION_PORT, port_name="emissionCO2")],
+    constraints=[Constraint(name='Bound CO2', expression=port_field("emissionCO2", "Q").sum_connections() <= param("quota"))]
+)
+
+"""
+Link model to interconnect nodes.
+"""
+LINK_MODEL = model(
+    id="LINK",
+    parameters=[float_parameter("f_max", TIME_AND_SCENARIO_FREE)],
+    variables=[
+        float_variable("flow", lower_bound=-param("f_max"), upper_bound=param("f_max"))
+    ],
+    ports=[
+        ModelPort(port_type=FLOW_PORT, port_name="port_from"),
+        ModelPort(port_type=FLOW_PORT, port_name="port_to"),
+    ],
+    port_fields_definitions=[
+        PortFieldDefinition(
+            port_field=PortFieldId("port_from", "F"),
+            definition=-var("flow"),
+        ),
+        PortFieldDefinition(
+            port_field=PortFieldId("port_to", "F"),
+            definition=var("flow"),
+        )
+    ]
+)
+
+
+"""
+build the quota CO² test system.
```

Comment:
Please provide a more detailed description of the test case, look at https://github.com/AntaresSimulatorTeam/andromede-modeling-prototype/blob/9fc446479def5d6f31a261281b59e8df4cf3f831/tests/andromede/test_andromede.py#L370 for an example